### PR TITLE
Fix aws token caching

### PR DESCRIPTION
**Pull request by anshubansal27**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Temporary Token is cached for the offloading and AWS refreshes the token around every 6 hours and offloading is getting failed with Expired Token Issue Error.

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

I am just removing the caching part of the token and every time calling the AWS API : refresh might be already handled inside AWS SDK code and therefore this additional check might not be required.

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@anshubansal27 thanks for your contribution. For this PR, do we need to update docs?

File: tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/provider/JCloudBlobStoreProvider.java [hunk 1]
```diff
@@ -279,47 +279,47 @@ public ProviderMetadata getProviderMetadata() {
     };
 
     static final CredentialBuilder AWS_CREDENTIAL_BUILDER = (TieredStorageConfiguration config) -> {
-        if (config.getCredentials() == null) {
-            final AWSCredentialsProvider authChain;
-            try {
-                if (Strings.isNullOrEmpty(config.getConfigProperty(S3_ROLE_FIELD))) {
-                    authChain = DefaultAWSCredentialsProviderChain.getInstance();
+//        if (config.getCredentials() == null) {
```
Comment (michaeljmarshall) [inline tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/provider/JCloudBlobStoreProvider.java:282]: Just a comment on style here: in a future PR, I recommend removing this line. Its record will be stored in git history, and there isn't any need to keep it for future reference.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
Thanks for your contribution @anshubansal27. 

A procedural note: PRs should target the `master` branch, and then once they're merged to `master`, you can request that the commit be cherry picked to an active branch, like `branch-2.7`.

Second, I think you'll need to provide additional justification for this change. I don't believe the proposed changes will remove caching as you suggest in your PR's description. Further, we want to cache tokens intelligently given that network calls have a cost and AWS has rate limits, so additional context motivating your PR would be very helpful to reviewers.

**Comment by michaeljmarshall** [comment]
@anshubansal27 - do you plan on providing an update to this PR?

**Comment by michaeljmarshall** [comment]
Closing for now due to inactivity. Please re-open if you're able to complete this contribution. Thanks!

**Pull request closed without merging**